engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/typer/TypeMethodReference.scala

```diff
@@ -20,8 +20,18 @@ object TypeMethodReference {
 class TypeMethodReference(methodReference: MethodReference, currentResults: List[TypingResult]) {
   def call: Either[String, TypingResult] =
     currentResults.headOption match {
-      case Some(Typed(typedClasses)) =>
-        typeFromClazzDefinitions(extractClazzDefinitions(typedClasses))
+      // TODO: make more understandable
+      case Some(tc: TypedClass) =>
+        typeFromClazzDefinitions(extractClazzDefinitions(Set(tc)))
+      case Some(Typed(nestedTypes)) =>
+        val typeClasses = nestedTypes.collect {
```

Comment:
I think in this case we should do flatMap instead

Reply:
And when there is non `TypedClass` in `nestedTypes` we should throw exception?

Reply:
Ok, after ensuring that `TypedUnion` contains only flat types (`ScalarTypingResult`) few things are easier now - e.g. in this place `collect` is not needed anymore.